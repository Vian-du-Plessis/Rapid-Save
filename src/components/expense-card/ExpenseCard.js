import React from 'react';
import styles from './ExpenseCard.module.scss';

import Icon from '../../assets/svg/close.svg';

const ExpenseCard = (props) => {
    return (
        <div className={ styles.container }>
            <div className={ styles.leftContainer }>
                <h4>{props.title}</h4>
                <h4>{props.price}</h4>
            </div>
            <div className={ styles.rightContainer }>
                <div 
                    className={ styles.iconContainer }
                    onClick={props.onClick}
                >
                    <img src={ Icon } alt="" /> 
                </div>
            </div>
        </div>
    );
};

export default ExpenseCard;