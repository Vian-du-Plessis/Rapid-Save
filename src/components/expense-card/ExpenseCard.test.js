import ExpenseCard from './ExpenseCard';
import "@testing-library/jest-dom/extend-expect";
import TestRenderer from 'react-test-renderer';
import { act, create } from "react-test-renderer";


describe("Test the ExpenseCard component ", () =>{
    test("Create Rendered ExpenseCard", () => {

        const testBtn = TestRenderer.create(
            <ExpenseCard
                title='Water'
                price='R 1000'
            />
        );
        const btnTestInstance = testBtn.root;
        expect(btnTestInstance.findByType(ExpenseCard).props.label).toBe["Water"];
    });

    test("ExpenseCard Matches Snapshot", async () => {
        let root;
        await act(() => {
            root = create(            
                <ExpenseCard
                    title='Water'
                    price='R 1000'
                />
            );
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            root.update(
                <ExpenseCard
                    title='Water'
                    price='R 1000'
                />
            );
            expect(root.toJSON()).toMatchSnapshot();
        })
    })
})
