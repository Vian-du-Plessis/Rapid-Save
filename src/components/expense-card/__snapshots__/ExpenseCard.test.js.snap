// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the ExpenseCard component  ExpenseCard Matches Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="leftContainer"
  >
    <h4>
      Water
    </h4>
    <h4>
      R 
      R 1000
    </h4>
  </div>
  <div
    className="rightContainer"
  >
    <div
      className="iconContainer"
    >
      <img
        alt=""
        src="close.svg"
      />
    </div>
  </div>
</div>
`;

exports[`Test the ExpenseCard component  ExpenseCard Matches Snapshot 2`] = `
<div
  className="container"
>
  <div
    className="leftContainer"
  >
    <h4>
      Water
    </h4>
    <h4>
      R 
      R 1000
    </h4>
  </div>
  <div
    className="rightContainer"
  >
    <div
      className="iconContainer"
    >
      <img
        alt=""
        src="close.svg"
      />
    </div>
  </div>
</div>
`;
